### PR TITLE
Fix sharing settings unit tests

### PR DESCRIPTION
**Pull request by artonge**
Follow up of https://github.com/nextcloud/server/pull/41581

### CODE REVIEW

**Review by susnux — APPROVED** [review]

File: apps/settings/tests/Settings/Admin/SharingTest.php [hunk 1]
```diff
@@ -52,20 +54,35 @@ class SharingTest extends TestCase {
...
 		$this->l10n = $this->getMockBuilder(IL10N::class)->getMock();
 
+		/** @var IManager|\PHPUnit\Framework\MockObject\MockObject */
 		$this->shareManager = $this->getMockBuilder(IManager::class)->getMock();
+		/** @var IAppManager|\PHPUnit\Framework\MockObject\MockObject */
 		$this->appManager = $this->getMockBuilder(IAppManager::class)->getMock();
+		/** @var IURLGenerator|\PHPUnit\Framework\MockObject\MockObject */
+		$this->urlGenerator = $this->getMockBuilder(IURLGenerator::class)->getMock();
+		/** @var IInitialState|\PHPUnit\Framework\MockObject\MockObject */
+		$this->initialState = $this->getMockBuilder(IInitialState::class)->getMock();
```
Comment (come-nc) [inline apps/settings/tests/Settings/Admin/SharingTest.php:76]:
```suggestion
		/** @var IConfig|MockObject */
		$this->config = $this->getMockBuilder(IConfig::class)->getMock();
		/** @var IL10N|MockObject */
		$this->l10n = $this->getMockBuilder(IL10N::class)->getMock();

		/** @var IManager|MockObject */
		$this->shareManager = $this->getMockBuilder(IManager::class)->getMock();
		/** @var IAppManager|MockObject */
		$this->appManager = $this->getMockBuilder(IAppManager::class)->getMock();
		/** @var IURLGenerator|MockObject */
		$this->urlGenerator = $this->getMockBuilder(IURLGenerator::class)->getMock();
		/** @var IInitialState|MockObject */
		$this->initialState = $this->getMockBuilder(IInitialState::class)->getMock();
```
nit

**Review by come-nc — APPROVED** [review]

**Comment by come-nc** [comment]
```
453	There were 2 failures:
454	
455	1) OCA\Settings\Tests\Settings\Admin\SharingTest::testGetFormWithoutExcludedGroups
456	Failed asserting that two objects are equal.
457	--- Expected
458	+++ Actual
459	@@ @@
460	 OCP\AppFramework\Http\TemplateResponse Object (
461	     'templateName' => 'settings/admin/sharing'
462	     'params' => Array (
463	-        'sharingAppEnabled' => false
464	-        'allowGroupSharing' => 'yes'
465	-        'allowLinks' => 'yes'
466	-        'allowPublicUpload' => 'yes'
467	-        'allowResharing' => 'yes'
468	-        'allowShareDialogUserEnumeration' => 'yes'
469	-        'restrictUserEnumerationToGroup' => 'no'
470	-        'restrictUserEnumerationToPhone' => 'no'
471	-        'restrictUserEnumerationFullMatch' => 'yes'
472	-        'restrictUserEnumerationFullMatchUserId' => 'yes'
473	-        'restrictUserEnumerationFullMatchEmail' => 'yes'
474	-        'restrictUserEnumerationFullMatchIgnoreSecondDN' => 'no'
475	-        'enforceLinkPassword' => false
476	-        'onlyShareWithGroupMembers' => false
477	-        'shareAPIEnabled' => 'yes'
478	-        'shareDefaultExpireDateSet' => 'no'
479	-        'shareExpireAfterNDays' => '7'
480	-        'shareEnforceExpireDate' => 'no'
481	-        'shareExcludeGroups' => false
482	-        'shareExcludedGroupsList' => ''
483	-        'publicShareDisclaimerText' => 'Lorem ipsum'
484	-        'enableLinkPasswordByDefault' => 'yes'
485	-        'shareApiDefaultPermissions' => 31
486	-        'shareApiDefaultPermissionsCheckboxes' => false
487	-        'shareDefaultInternalExpireDateSet' => 'no'
488	-        'shareInternalExpireAfterNDays' => '7'
489	-        'shareInternalEnforceExpireDate' => 'no'
490	-        'shareDefaultRemoteExpireDateSet' => 'no'
491	-        'shareRemoteExpireAfterNDays' => '7'
492	-        'shareRemoteEnforceExpireDate' => 'no'
493	-        'allowLinksExcludeGroups' => ''
494	-        'passwordExcludedGroups' => ''
495	-        'passwordExcludedGroupsFeatureEnabled' => false
496	     )
497	     'renderAs' => ''
498	-    'appName' => 'settings'
499	+    'appName' => ''
500	     'headers' => Array ()
501	     'cookies' => Array ()
502	     'status' => 200
503	
504	/drone/src/apps/settings/tests/Settings/Admin/SharingTest.php:168
505	
506	2) OCA\Settings\Tests\Settings\Admin\SharingTest::testGetFormWithExcludedGroups
507	Failed asserting that two objects are equal.
508	--- Expected
509	+++ Actual
510	@@ @@
511	 OCP\AppFramework\Http\TemplateResponse Object (
512	     'templateName' => 'settings/admin/sharing'
513	     'params' => Array (
514	-        'sharingAppEnabled' => true
515	-        'allowGroupSharing' => 'yes'
516	-        'allowLinks' => 'yes'
517	-        'allowPublicUpload' => 'yes'
518	-        'allowResharing' => 'yes'
519	-        'allowShareDialogUserEnumeration' => 'yes'
520	-        'restrictUserEnumerationToGroup' => 'no'
521	-        'restrictUserEnumerationToPhone' => 'no'
522	-        'restrictUserEnumerationFullMatch' => 'yes'
523	-        'restrictUserEnumerationFullMatchUserId' => 'yes'
524	-        'restrictUserEnumerationFullMatchEmail' => 'yes'
525	-        'restrictUserEnumerationFullMatchIgnoreSecondDN' => 'no'
526	-        'enforceLinkPassword' => false
527	-        'onlyShareWithGroupMembers' => false
528	-        'shareAPIEnabled' => 'yes'
529	-        'shareDefaultExpireDateSet' => 'no'
530	-        'shareExpireAfterNDays' => '7'
531	-        'shareEnforceExpireDate' => 'no'
532	-        'shareExcludeGroups' => true
533	-        'shareExcludedGroupsList' => 'NoSharers|OtherNoSharers'
534	-        'publicShareDisclaimerText' => 'Lorem ipsum'
535	-        'enableLinkPasswordByDefault' => 'yes'
536	-        'shareApiDefaultPermissions' => 31
537	-        'shareApiDefaultPermissionsCheckboxes' => false
538	-        'shareDefaultInternalExpireDateSet' => 'no'
539	-        'shareInternalExpireAfterNDays' => '7'
540	-        'shareInternalEnforceExpireDate' => 'no'
541	-        'shareDefaultRemoteExpireDateSet' => 'no'
542	-        'shareRemoteExpireAfterNDays' => '7'
543	-        'shareRemoteEnforceExpireDate' => 'no'
544	-        'allowLinksExcludeGroups' => ''
545	-        'passwordExcludedGroups' => ''
546	-        'passwordExcludedGroupsFeatureEnabled' => false
547	     )
548	     'renderAs' => ''
549	-    'appName' => 'settings'
550	+    'appName' => ''
551	     'headers' => Array ()
552	     'cookies' => Array ()
553	     'status' => 200
554	
555	/drone/src/apps/settings/tests/Settings/Admin/SharingTest.php:250
```

File: apps/settings/tests/Settings/Admin/SharingTest.php [hunk 2]
```diff
@@ -106,45 +123,51 @@ public function testGetFormWithoutExcludedGroups(): void {
 			->willReturn(false);
 
 		$this->appManager->method('isEnabledForUser')->with('files_sharing')->willReturn(false);
+		$this->initialState
```
Comment (come-nc) [inline apps/settings/tests/Settings/Admin/SharingTest.php:126]: Missing an expect, no?

Comment (artonge) [reply apps/settings/tests/Settings/Admin/SharingTest.php:126]: Well, I do not think so :thinking:

Comment (artonge) [reply apps/settings/tests/Settings/Admin/SharingTest.php:126]: Just checked other tests, and you are right, updated.

File: apps/settings/tests/Settings/Admin/SharingTest.php [hunk 3]
```diff
@@ -106,45 +123,52 @@ public function testGetFormWithoutExcludedGroups(): void {
...
+			->expect($this->once())
+			->method('provideInitialState')->w
```
Comment (come-nc) [inline apps/settings/tests/Settings/Admin/SharingTest.php:128]: Typo?

**Comment by come-nc** [comment]
```
PHP Fatal error:  Uncaught ParseError: syntax error, unexpected token "->", expecting identifier or variable or "{" or "$" in /drone/src/apps/settings/tests/Settings/Admin/SharingTest.php:129
```
:facepalm:

**Comment by come-nc** [comment]
Fixed it until it passed locally :disappointed_relieved:

**Pull request merged**